pkg/protocol/protocol.go

```diff
@@ -284,14 +285,11 @@ func (p *Protocol) ProcessBlock(block *model.Block, src network.PeerID) error {
 		processed = true
 	}
 
-	if candidateEngine := p.CandidateEngineInstance(); candidateEngine != nil {
-		if candidateChain := candidateEngine.ChainID(); chainCommitment.Chain().ForkingPoint.ID() == candidateChain || candidateEngine.BlockRequester.HasTicker(block.ID()) {
-			candidateEngine.ProcessBlockFromPeer(block, src)
-			if candidateEngine.IsBootstrapped() &&
-				candidateEngine.Storage.Settings().LatestCommitment().CumulativeWeight() > mainEngine.Storage.Settings().LatestCommitment().CumulativeWeight() {
-				// TODO: when activating the candidate engine should we make sure that the candidate engine produces the same commitments as the chain we decided to switch to?
-				//   -> if they are different we could shut down this engine.
-
+	if candidateEngineInstance := p.CandidateEngineInstance(); candidateEngineInstance != nil {
+		if candidateChain := candidateEngineInstance.ChainID(); chainCommitment.Chain().ForkingPoint.ID() == candidateChain || candidateEngineInstance.BlockRequester.HasTicker(block.ID()) {
+			candidateEngineInstance.ProcessBlockFromPeer(block, src)
+			if candidateEngineInstance.IsBootstrapped() &&
+				candidateEngineInstance.Storage.Settings().LatestCommitment().CumulativeWeight() > mainEngine.Storage.Settings().LatestCommitment().CumulativeWeight() {
```

Comment:
How do you enforce here that the candidateEngine is actually synced on the chain you meant to sync in the first place?
Why shouldn't I be able to prove to you that a heavier chain exists, and then make your candidateEngine sync on a very shallow fork (1 commitment deep) that is heavier than your main engine? You will switch here!